### PR TITLE
Adding `arraymap` package

### PR DESCRIPTION
**Pull request by flexatone**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by flexatone** [comment]
I am willing to be a recipe maintainer.

**Comment by chaburkland** [comment]
I am willing to be a recipe maintainer

**Comment by flexatone** [comment]
@conda-forge/help-python-c, ready for review!

**Comment by flexatone** [comment]
https://github.com/orgs/conda-forge/teams/help-python-c, ready for review!

**Comment by flexatone** [comment]
@conda-forge/help-python , ready for review!

File: recipes/arraymap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy ==1.18.5  # [py==37]
+    - numpy ==1.18.5  # [py==38]
+    - numpy >=1.18.5  # [py>=39]
```
Comment (xhochy) [inline recipes/arraymap/meta.yaml:25]: Don't constrain this so it uses the global pinning.
```suggestion
    - numpy
```

File: recipes/arraymap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  run:
+    - python
+    - numpy ==1.18.5  # [py==37]
+    - numpy ==1.18.5  # [py==38]
+    - numpy >=1.18.5  # [py>=39]
```
Comment (xhochy) [inline recipes/arraymap/meta.yaml:30]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/arraymap/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 16edc1b39cb66d62c6e2561feb236371f7819fcbe5225200d1a4663cc79568a3
+
+build:
+  number: 0
+  skip: true  # [py<37]
```
Comment (xhochy) [inline recipes/arraymap/meta.yaml:14]: Not needed anymore 
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by flexatone** [comment]
Thanks, @xhochy for the corrections, much appreciated!

**Review by xhochy — APPROVED** [review]

**Pull request merged**